components/server/src/util/log-context.ts

```diff
@@ -64,3 +65,19 @@ export function wrapAsyncGenerator<T>(
         },
     };
 }
+
+export function getZedTokenFromContext(): StoredZedToken | undefined {
+    return asyncLocalStorage.getStore()?.zedToken;
+}
+export function setZedTokenToContext(zedToken: StoredZedToken) {
+    const ctx = asyncLocalStorage.getStore();
+    if (ctx) {
+        ctx.zedToken = zedToken;
```

Comment:
does it really work? I thought the only way to modify the storage is to use `run`?

Reply:
My understanding was that this will work when called from code that runs inside `runWithContext`. As we run all API calls inside, I thought it should.
Might be wrong, though. :upside_down_face: 

/cc @svenefftinge

Reply:
just checking, I don't know whether you can mutate, or only read

Reply:
> Just not super cool to hijack the "log"-context here. Do we want to move the code that sets the asyncLocalStorage out to something more general and make use of that from logging?

Totally agree. Would it be ok to merge this now (so we get some data), and me working on this in parallel?